src/TBuffer.cpp

```diff
@@ -5156,3 +5188,49 @@ int TBuffer::lengthInGraphemes(const QString& text)
     }
     return count;
 }
+
+const QList<QByteArray> TBuffer::getEncodingNames()
+{
+    static QByteArrayList results;
+    if (results.isEmpty()) {
+        // First call, list is empty - so work them out, just this once:
+        results = QByteArrayList{csmEncodingTable.keys()};
+
+        QMutableByteArrayListIterator itEncoding(results);
+        while (itEncoding.hasNext()) {
+            QByteArray encoding{itEncoding.next()};
+            QTextCodec* pEncoding = QTextCodec::codecForName(encoding);
+            if (!pEncoding) {
+                // We do not have that encoder available after all
+                itEncoding.remove();
+                if (encoding == "CP437") {
+                    // Okay to insert our replacement TTextCodex_XXXX into the
+                    // system we must instantiate them once:
+                    auto* pTTextCodec_437 = new TTextCodec_437();
```

Comment:
Could you change this to be a `std::unique_ptr`? Raw pointers and `new` can be error-prone - [modern way](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-smart) is to use smart pointers. if someone else changes this code and misses a `delete`, they'd make a memory leak - so we're in a way setting up a trap for them.

Reply:
NO! You must NOT `delete` the object that these pointers to `TTextCodec` instances point at - or arrange for that to happen with a fancy smart-assed pointer. The internal Qt codec managing infra-structure takes ownership of them once they are created and used and will delete them at the appropriate time itself.

Reply:
ok!

Reply:
I did have a `Q_UNUSED(...)` around the pointer in the first version - as I wasn't using it then - with a sarky comment about how it was used by Qt and not to `delete` it. Guess I need to put something back in about that...

Reply:
Humm, it looks like I already *had*. That is followed by:
```CPP
                    // Now that it has been instantiated, the system knows about
                    // it - indeed it takes possession of it and we must NOT
                    // delete it ourselves!
```